gel/_internal/_qbmodel/_abstract/_base.py

```diff
@@ -162,8 +165,13 @@ def __new__(
             super().__new__(mcls, name, bases, namespace, **kwargs),
         )
         reflection = cls.__gel_reflection__
-        if (tname := getattr(reflection, "name", None)) is not None:
-            mcls.__gel_class_registry__[tname] = cls
+        if (
+            (tname := getattr(reflection, "name", None)) is not None
+            and not issubclass(cls, AbstractGelLinkModel)
+            and not hasattr(cls, '__linkprops__')
+            and not hasattr(cls, '__gel_merged_model__')
```

Comment:
It is probably more correct to check for `__gel_root_class__` which we should uplift from `_pydantic.GelModelMeta`.

Reply:
I changed this mechanism to check for the presence of `tname__` in namespace, which picks up all of these cases more cleanly and covers two cases I had missed `__build_custom_serializer` and user-defined subtypes